src/libfuncs/array.rs

```diff
@@ -1648,11 +1653,9 @@ pub fn build_tuple_from_span<'ctx, 'this>(
             &info.signature.param_signatures[0].ty,
         )?;
 
-        // The following unwrap is unreachable because an array always has a drop implementation,
-        // which at this point is always inserted thanks to the `build_type()` just above.
```

Comment:
This comment gives valuable information on why the following operation will never fail. I think you should keep it, even though you now replaced the unwrap for a proper error. You could rephrase it like this: "the following operation will never fail because..."

Reply:
Done!